destination/clickhouse.go

```diff
@@ -30,146 +31,185 @@ type ClickHouseConnection struct {
 }
 
 func GetClickHouseConnection(ctx context.Context, configuration map[string]string) (*ClickHouseConnection, error) {
-	hostname := fmt.Sprintf("%s:%s",
-		GetWithDefault(configuration, "hostname", "localhost"),
-		GetWithDefault(configuration, "port", "9000"))
-	username := GetWithDefault(configuration, "username", "default")
-	password := GetWithDefault(configuration, "password", "")
-	database := GetWithDefault(configuration, "database", "default")
-	options := &clickhouse.Options{
-		Addr: []string{hostname},
-		Auth: clickhouse.Auth{
-			Username: username,
-			Password: password,
-			Database: database,
-		},
-		Protocol: clickhouse.Native,
-		Settings: clickhouse.Settings{
-			"allow_experimental_object_type": 1,
-			"date_time_input_format":         "best_effort",
-			//"async_insert":                   1,
-			//"wait_for_async_insert":          1,
-		},
-		ClientInfo: clickhouse.ClientInfo{
-			Products: []struct {
-				Name    string
-				Version string
-			}{
-				{Name: "fivetran-destination", Version: Version},
+	return RetryNetErrorWithData(func() (*ClickHouseConnection, error) {
+		hostname := fmt.Sprintf("%s:%s",
+			GetWithDefault(configuration, "hostname", "localhost"),
+			GetWithDefault(configuration, "port", "9000"))
+		username := GetWithDefault(configuration, "username", "default")
+		password := GetWithDefault(configuration, "password", "")
+		database := GetWithDefault(configuration, "database", "default")
+		options := &clickhouse.Options{
+			Addr: []string{hostname},
+			Auth: clickhouse.Auth{
+				Username: username,
+				Password: password,
+				Database: database,
 			},
-		},
-	}
-	ssl := GetWithDefault(configuration, "ssl", "false")
-	if ssl == "true" {
-		options.TLS = &tls.Config{InsecureSkipVerify: true}
-	}
-	conn, err := clickhouse.Open(options)
-	if err != nil {
-		return nil, err
-	}
-	return &ClickHouseConnection{database, ctx, conn}, nil
+			Protocol: clickhouse.Native,
```

Comment:
Is there any reason to not support HTTP as well?

Reply:
Huge SELECTs for each update/delete batch (may cause max request size issues) and potential issues with KeepAlive that I want to avoid.

Reply:
> Huge SELECTs for each update/delete batch (may cause max request size issues)

Is it a real issue? E.g. we don't experience it in ClickPipes.

> and potential issues with KeepAlive that I want to avoid.

same as above.

Reply:
@jkaflik, apologies for the confusion, I was talking about SELECT _query_ size, not the result set.
```sql
SELECT * FROM `foo`.`bar` FINAL 
WHERE (id, name) IN (
  (42, 'foo'), 
  (43, 'bar'), 
  (44, 'qaz'), 
  (45, 'qux'),
  -- and so on, eventually becomes quite long
) ORDER BY (id, name)
```